openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -177,10 +179,11 @@ def set_user_preference(requesting_user, preference_key, preference_value, usern
     existing_user = _get_authorized_user(requesting_user, username)
     serializer = create_user_preference_serializer(existing_user, preference_key, preference_value)
     validate_user_preference_serializer(serializer, preference_key, preference_value)
-    try:
-        serializer.save()
-    except Exception as error:
-        raise _create_preference_update_error(preference_key, preference_value, error)
+    if serializer.instance is None or serializer.instance.value != serializer.validated_data['value']:
+        try:
+            serializer.save()
+        except Exception as error:
```

Comment:
To some degree: not my code. But also, I think the goal is to make it so that the set of exceptions thrown by this api is limited, hence wrapping all of the ones coming from the underlying system.